### DISCUSSION TITLE
Swift 类构造器的使用 · /swift-zhong-init-de-shi-yong

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#497] [Announcements]
https://draveness.me/swift-zhong-init-de-shi-yong

**Comment by magicer0126** [comment]
大佬，代码有点少，文字多，需要不时的往上翻。swift 构造器调用规则总结的非常棒。

**Comment by dagundejuzi** [comment]
构造器在第一阶段构造完成之前, 不能调用任何实例方法, 不能读取任何实例属性的值，self 不能被引用. 
这条规则似乎和错误5的示例代码不符？
